sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentField.cs

```diff
@@ -311,5 +326,23 @@ public AddressValue AsAddress()
 
             return ValueAddress;
         }
+
+        private object InternalValue => ValueType switch
+        {
+            DocumentFieldType.Address => ValueAddress,
+            DocumentFieldType.CountryRegion => ValueCountryRegion,
+            DocumentFieldType.Currency => ValueCurrency,
```

Comment:
if I understand correctly, if the property has a value, the user can see it here.
If it is null, the user will see null right? And then they'll try to access it and then we will throw.
Is this the desired behavior?

Reply:
That's a good point.

After discussing with the service team, they confirmed they can't guarantee the `Value{...}` private properties will always be returned, so we should update our code to reflect the possibility of them being null. This reminds me of some negative feedback we got a couple of times about the conversion methods (`AsDate`, `AsTime`, etc) throwing exceptions customers were not expecting, and about considering implementing them with the `Try` pattern in .NET (such as in `TryConvertToInt64`).

From my perspective, the implementation of `InternalValue` is more aligned to what we want for GA, and conversion methods should be updated somehow. I agree that having them behaving differently can lead to confusion. What do you think about updating `InternalValue` to rely on the current conversion methods so they have the same behavior for now? For example:
```cs
        private object InternalValue => ValueType switch
        {
            DocumentFieldType.Address => AsAddress(),
            DocumentFieldType.CountryRegion => AsCountryRegion(),
            DocumentFieldType.Currency => AsCurrency(),
            // ....
```

Reply:
> What do you think about updating InternalValue to rely on the current conversion methods so they have the same behavior for now

Agree and as you said, we can test this out on the beta release and keep looking into it before GA. It would be good ot have an issue tracking the feedback so that this gets look at before GA

Reply:
We have an open issue for it: https://github.com/Azure/azure-sdk-for-net/issues/29013
I'm not closing it with this PR since I can't say this is a final design yet. `DocumentField` will change a bit until GA.